File: Samples/DecodeTile/VB.NET/DecodeTile.vb
Imports BitMiracle.Jpeg2k

Module DecodeTile

    Sub Main()
        ' The library won't work without a license. You can get free time limited license
        ' at https://bitmiracle.com/jpeg2000/
        LicenseManager.SetTrialLicense("contact support@bitmiracle.com for a license")

        Dim fileName As String = "Sample data/p1_04.j2k"
        Dim tileFileName As String = "tile"

        Using image = New J2kImage(fileName)
            Dim options = New J2kDecodingOptions()
            options.ForceRgbColorSpace = False

            For Each i As Integer In New Integer() {5, 25, 45}
                Dim imageData = image.DecodeTile(i, options)
                Dim outputName As String = String.Format("{0}_{1}.tif", tileFileName, i)
                imageData.Save(outputName, J2kOutputFormat.Tiff)

                Process.Start(outputName)
            Next
        End Using
    End Sub

End Module

File: Samples/DecodeTile/C#/DecodeTile.cs
using BitMiracle.Jpeg2k;
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Text;

namespace DecodeTile
{
    class DecodeTile
    {
        static void Main(string[] args)
        {
            // The library won't work without a license. You can get free time limited license
            // at https://bitmiracle.com/jpeg2000/
            LicenseManager.SetTrialLicense("contact support@bitmiracle.com for a license");

            string fileName = @"Sample data/p1_04.j2k";
            string tileFileName = @"tile";

            using (var image = new J2kImage(fileName))
            {
                var options = new J2kDecodingOptions
                {
                    ForceRgbColorSpace = false
                };

                foreach (int i in new int[] { 5, 25, 45} )
                {
                    var imageData = image.DecodeTile(i, options);
                    string outputName = string.Format("{0}_{1}.tif", tileFileName, i);
                    imageData.Save(outputName, J2kOutputFormat.Tiff);

                    Process.Start(outputName);
                }
            }
        }
    }
}
